### PR TITLE
Minor bug fixes and features adjustments.

### LINKED ISSUE #52: Program flag "PatronReviewFlag" doesn't seem to do anything.
I was unsure what the "PatrongReviewFlag" did. After a search it doesn't appear to affect the program at all. Am I missing something?

Here is a list of usages:

> Find all "PatronReviewFlag", Find Results 1, Entire Solution, ""
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\CreateSchema.sql(10366):	@PatronReviewFlag BIT,
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\CreateSchema.sql(10417):		PatronReviewFlag,
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\CreateSchema.sql(10468):		@PatronReviewFlag,
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\CreateSchema.sql(10631):	@PatronReviewFlag BIT,
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\CreateSchema.sql(10680):	PatronReviewFlag = @PatronReviewFlag,
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\CreateSchema.sql(21720):	[PatronReviewFlag] [bit] NULL,
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\InsertInitialData-MultiplePrograms.sql(1685):	[PatronReviewFlag],
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\InsertInitialData-MultiplePrograms.sql(1785):	[PatronReviewFlag],
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\InsertInitialData-MultiplePrograms.sql(1885):	[PatronReviewFlag],
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\InsertInitialData-MultiplePrograms.sql(1985):	[PatronReviewFlag],
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Install\InsertInitialData.sql(1685):	[PatronReviewFlag],
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Programs\ProgramsAddEdit.aspx(409):                                            <asp:CheckBox ID="PatronReviewFlag" runat="server" Checked='<%# (bool)Eval("PatronReviewFlag") %>' ReadOnly="False"></asp:CheckBox>
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Programs\ProgramsAddEdit.aspx.cs(124):                    //obj.PatronReviewFlag = ((CheckBox)((DetailsView)sender).FindControl("PatronReviewFlag")).Checked;
  C:\cygwin64\home\justinm\develop\gra\SRP\ControlRoom\Modules\Programs\ProgramsAddEdit.aspx.cs(216):                    obj.PatronReviewFlag = ((CheckBox)((DetailsView)sender).FindControl("tc1").FindControl("tp1").FindControl("PatronReviewFlag")).Checked;
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(44):        private bool myPatronReviewFlag;
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(159):        public bool PatronReviewFlag
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(161):            get { return myPatronReviewFlag; }
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(162):            set { myPatronReviewFlag = value; }
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(493):                result.PatronReviewFlag = bool.Parse(dr["PatronReviewFlag"].ToString());
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(584):                this.PatronReviewFlag = bool.Parse(dr["PatronReviewFlag"].ToString());
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(706):            arrParams[14] = new SqlParameter("@PatronReviewFlag", GRA.SRP.Core.Utilities.GlobalUtilities.DBSafeValue(o.PatronReviewFlag, o.PatronReviewFlag.GetTypeCode()));
  C:\cygwin64\home\justinm\develop\gra\SRP_DAL\Programs.cs(782):            arrParams[15] = new SqlParameter("@PatronReviewFlag", GRA.SRP.Core.Utilities.GlobalUtilities.DBSafeValue(o.PatronReviewFlag, o.PatronReviewFlag.GetTypeCode()));
  C:\cygwin64\home\justinm\develop\gra\GRA.Database\dbo\Stored Procedures\app_Programs_Insert.sql(17):	@PatronReviewFlag BIT,
  C:\cygwin64\home\justinm\develop\gra\GRA.Database\dbo\Stored Procedures\app_Programs_Insert.sql(68):		PatronReviewFlag,
  C:\cygwin64\home\justinm\develop\gra\GRA.Database\dbo\Stored Procedures\app_Programs_Insert.sql(119):		@PatronReviewFlag,
  C:\cygwin64\home\justinm\develop\gra\GRA.Database\dbo\Stored Procedures\app_Programs_Update.sql(18):	@PatronReviewFlag BIT,
  C:\cygwin64\home\justinm\develop\gra\GRA.Database\dbo\Stored Procedures\app_Programs_Update.sql(67):	PatronReviewFlag = @PatronReviewFlag,
  C:\cygwin64\home\justinm\develop\gra\GRA.Database\dbo\Tables\Programs.sql(17):    [PatronReviewFlag]    BIT           NULL,
  Matching lines: 28    Matching files: 9    Total files searched: 1591

### LINKED ISSUE #151: Reading goal should only use points earned by reading.
(assign to me)

**Expected**
Reading points earn progress towards reading goal.

**Actual**
All points, regardless of source result in progress towards reading goal.

**Description**
This is a misunderstanding on my part on how points were earned. I did not realize that points could come from so many sources, this is misleading because it allows progress to be recorded on the "reading goal" without reading. Only patron points from reading should be counted towards the goal. "Overall goal" label should probably be changed to "reading goal".

### LINKED ISSUE #153: Avatar editor says "Award Name" should say "Avatar Name"
Assign to me

### PR DESCRIPTION
**Pull request by justinmeiners**
- [x] Ready for merge
- [x] Fixed #153, "Award Name" should say "Avatar Name"
- [x] Fixed #151, "Reading goal should only use points earned by reading"
- [x] Fixed #52, Program flag "Allow Patron to enter book review" now provides field.
- [x] Patron registration process now informs user about goal range requirements.

### CODE REVIEW

**Comment by haraldnagel** [comment]
Hey, one other note as you work on this: I added a `CHANGELOG.md` document to the root of the project (and I added it to the "Solution Items" folder in Visual Studio but I'm not sure if that change has made it back into MCLD/master) - I'm trying to follow the [change log guidance at keepachangelog.com](http://keepachangelog.com/) since we're not doing the big-honking-pull-request approach now. Just ensure you add entries for all your adds/changes/removals/fixes/etc. :+1:

**Comment by justinmeiners** [comment]
I will go ahead and do that.

**Comment by justinmeiners** [comment]
If my changes look acceptable to you they should be ready to merge. I just went through a test with a fresh installation to ensure the database changes were made appropriately and the reading goal triggers work correctly.

File: SRP/Controls/PatronRegistration.ascx [hunk 1]
```diff
@@ -494,6 +504,11 @@
                                 <asp:RequiredFieldValidator runat="server" Enabled='<%# Eval("LibraryCard_Req") %>'
                                     ControlToValidate="LibraryCard" Display="Dynamic" ErrorMessage="Library card # is required"
                                     SetFocusOnError="True">required</asp:RequiredFieldValidator>
+
+                                <asp:RegularExpressionValidator runat="server" ControlToValidate="LibraryCard" Display="Dynamic"
+                                    ErrorMessage="Library card number is invalid."
+                                    SetFocusOnError="True"
+                                    ValidationExpression="^((21391|23005)(\d{9}))($|\s*)">invalid</asp:RegularExpressionValidator>
```
Comment (haraldnagel) [inline SRP/Controls/PatronRegistration.ascx]: Hmm, I think you need to pull this block out. :smile:

File: SRP/Web.config [hunk 2]
```diff
@@ -18,16 +18,11 @@
   </connectionStrings>
```
Comment (haraldnagel) [inline SRP/Web.config]: I think you probably want to back out these `Web.config` changes.

**Comment by haraldnagel** [comment]
This looks pretty good! Let me know when you address those two notes and we'll :shipit:

**Comment by justinmeiners** [comment]
I must have made a mistake in my branch merging because I specifically checked that! Thank you for the catch. The commit is now updated.

**Pull request merged**